core/src/main/java/org/testcontainers/containers/CommandExecutor.java

```diff
@@ -0,0 +1,82 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.ExecCreateCmdResponse;
+import org.slf4j.Logger;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.output.FrameConsumerResultCallback;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.ToStringConsumer;
+import org.testcontainers.utility.TestEnvironment;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+public interface CommandExecutor extends ContainerState {
```

Comment:
While I understand the motivation for this interface (and also LogFollower), I see more value in some utility class with `execInContainer(containerId, Charset charset, String...command)` method instead, and both GenericContainer and WaitStrategy calling it when needed.

The motivation is simple:
1) it will be possible to use it with **any** container (currently one will have to implement this interface to access this functionality)
2) it will get us closer to the "reusable container patterns" idea we had some time ago
3) having less interfaces gives us more flexibility to change the API in future

So, something like:
```java
public class ExecInContainerPattern {

    // Note generic ExecResult instead of Container.ExecResult (should be converted to later in GenericContainer)
    public static ExecResult execInContainer(String containerId, Charset charset, String... command) {
        // ...
    }
}
```

WDYT?

Reply:
Made this change along with the other suggested changes in the latest commit.
I've added a `Logger` parameter to the method in this new class, it's probably not ideal since it is only used for debug and trace messages, but they can be useful and I didn't want to change the existing behaviour.